src/lm.jl

```diff
@@ -123,14 +123,21 @@ function StatsBase.fit!(obj::LinearModel)
 end
 
 function fit(::Type{LinearModel}, X::AbstractMatrix{<:Real}, y::AbstractVector{<:Real},
-             allowrankdeficient::Bool=false; wts::AbstractVector{<:Real}=similar(y, 0))
+             allowrankdeficient_dep::Union{Bool,Nothing}=nothing;
+             wts::AbstractVector{<:Real}=similar(y, 0),
+             allowrankdeficient::Bool=false)
+    if allowrankdeficient_dep != nothing
```

Comment:
This helps the compiler to completely drop this code when the argument is not `nothing`.
```suggestion
    if allowrankdeficient_dep !== nothing
```